libimage/filters.go

```diff
@@ -48,115 +66,175 @@ func filterImages(images []*Image, filters []filterFunc) ([]*Image, error) {
 // compileImageFilters creates `filterFunc`s for the specified filters.  The
 // required format is `key=value` with the following supported keys:
 //           after, since, before, containers, dangling, id, label, readonly, reference, intermediate
-func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOptions) ([]filterFunc, error) {
+func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOptions) (disjoint, filters []filterFunc, err error) {
```

Comment:
Can you turn the return value into one map with `map[string][]filterFunc`?  I think that will simplify the logic.